selenium_tests/base_page.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from functools import partial
+
+from selenium.common.exceptions import (
+    NoSuchElementException,
+    InvalidSwitchToTargetException
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class BasePageMeta(type):
+    """
+    This metaclass allows us to avoid writing boilerplate that just finds
+    an element in a page by a locator tuple.
+
+    Expected usage:
+
+        class SomePageLocators(object):
+            TITLE = (By.CSS_SELECTOR, '#title')
+            BUTTON = (By.CSS_SELECTOR, '#the_only_button')
+
+        class SomePage(BasePage):
+            locator_class = SomePageLocators
+            located_properties = ['title', 'button']
+
+    This metaclass adds properties for each name in located_properties,
+    using the corresponding (upper-cased) attribute on the locator_class.
+    In the example above, it would add SomePage.title, which would look
+    up the element found by SomePageLocators.TITLE, and SomePage.button,
+    returning the element found by SomePageLocators.BUTTON.
+    """
+    def __new__(class_, name, bases, dict_):
```

Comment:
It's not a bad idea, actually...put it on the todo list for when we split this out into a separate project.